packages/types/src/transaction/view-transaction-plan/view-action-plan.test.ts

```diff
@@ -0,0 +1,286 @@
+    test('includes the amount', () => {
+      const actionView = viewActionPlan(metadataByAssetId)(validSpendActionPlan);
+      const spendView = actionView.actionView.value as SpendView;
+      const spendViewVisible = spendView.spendView.value as SpendView_Visible;
+
+      expect(spendViewVisible.note!.value?.valueView.value?.amount).toEqual({
+        hi: 1n,
+        lo: 0n,
+      });
+    });
+
+    test('throws if the amount is missing', () => {
+      const actionPlan = new ActionPlan({
+        action: {
+          case: 'spend',
+          value: {
+            note: {
+              address: { altBech32m: address },
+            },
+          },
+        },
+      });
+
+      expect(() => viewActionPlan({})(actionPlan)).toThrow('No value in note');
+    });
+
+    test('includes the denom metadata', () => {
+      const actionView = viewActionPlan(metadataByAssetId)(validSpendActionPlan);
+      const spendView = actionView.actionView.value as SpendView;
+      const spendViewVisible = spendView.spendView.value as SpendView_Visible;
+      const valueView = spendViewVisible.note!.value?.valueView.value as ValueView_KnownDenom;
+
+      expect(valueView.denom?.toJson()).toEqual(denomMetadata.toJson());
+    });
+
+    test('throws if the asset ID is missing', () => {
+      const actionPlan = new ActionPlan({
+        action: {
+          case: 'spend',
+          value: {
+            note: {
+              address: { altBech32m: address },
+              value: { amount: { hi: 1n, lo: 0n } },
+            },
+          },
+        },
+      });
+
+      expect(() => viewActionPlan({})(actionPlan)).toThrow('No asset ID in value');
+    });
+
+    test('throws if the asset ID refers to an unknown asset type', () => {
+      const actionPlan = new ActionPlan({
+        action: {
+          case: 'spend',
+          value: {
+            note: {
+              address: { altBech32m: address },
+              value: { amount: { hi: 1n, lo: 0n }, assetId: { altBech32m: 'invalid' } },
+            },
+          },
+        },
+      });
+
+      expect(() => viewActionPlan({})(actionPlan)).toThrow(
+        'Asset ID in spend plan refers to an unknown asset type',
+      );
+    });
+  });
+
+  describe('`output` action', () => {
+    const destAddress = new Address({ altBech32m: address });
+    const validOutputActionPlan = new ActionPlan({
+      action: {
+        case: 'output',
+        value: {
+          value: {
+            amount: { hi: 1n, lo: 0n },
+            assetId,
+          },
+          destAddress,
+        },
+      },
+    });
+
+    test('includes the destAddress', () => {
+      const actionView = viewActionPlan(metadataByAssetId)(validOutputActionPlan);
+      const outputView = actionView.actionView.value as OutputView;
+      const outputViewVisible = outputView.outputView.value as OutputView_Visible;
+
+      expect(outputViewVisible.note?.address?.addressView.value?.address).toEqual(destAddress);
+    });
```

Comment:
See [this comment](https://github.com/penumbra-zone/web/pull/366#discussion_r1451209146). Not sure if the `destAddress` should be put into the note... there's kind of a weird structure to the `output` action, where it has top-level `destAddress` and `value` keys, which I _think_ (but am not sure) should go in the `OutputView`'s `note` property... but I'm not sure if that's an accurate representation of the output action...?

Reply:
did you find an answer on this?

Reply:
Yep, Henry answered it above.